runtime/tests/org.eclipse.core.tests.runtime/src/org/eclipse/core/tests/runtime/jobs/IJobManagerTest.java

```diff
@@ -1235,6 +1235,7 @@ protected IStatus run(IProgressMonitor monitor) {
 						public boolean aboutToWait(Thread lockOwner) {
 							// aboutToWait will be called when main job will start joining the running job
 							if (!scheduled) {
+								running.terminate();
```

Comment:
i don't get this change. The "running" Job was seems intended to be run without burning any time. you only "need" to terminate it here because you increased the time to spend. Whats wrong with a short running job? can it be too short?

Reply:
The [original bug report](https://bugs.eclipse.org/bugs/show_bug.cgi?id=403271#c0:), for which this is a regression test, states the following:
> There are three jobs required to reproduce the problem:
> a) main job that calls manager.join(..)
> a) running job for which the main job is waiting for
> c) waiting job scheduled when the job manager is suspended and the main job is already joining the running job

The problem is that the execution time of the "running" job is so low that it may terminate before `join` is called in the main thread. Then the `aboutToWait` method of the lock listener is not called, because `join` does not lock at all, so the "waiting" job will not be scheduled (and part (c) of the reproduction is not fulfilled). This is exactly what happens when  the test case is executed on lower-performance infrastructure, such as for example in https://github.com/eclipse-platform/eclipse.platform/actions/runs/4573426969.

The proposed change ensures that the "running" job is actually still running when `join` is called by the main thread to ensure that the lock listener gets called. Then the "running" thread can safely be terminated and the waiting job can be scheduled. The `join` call will unblock because the "running" job terminates and the "waiting" job will be scheduled as expected.

_Side note_: These short running jobs are intended to ensure that the job is running when reaching a specific statement in a test case, but then to terminate as soon as possible to avoid burning time. In consequence, basically all short running jobs in tests have a risk to produce random failures (and will if they are executed on an infrastructure with according performance properties). The changes proposed here solve this kind of problem (for the recently failing test cases) by having a long running job that is terminated exactly when it is expected to in terms of executed statements rather than elapsed time.

Reply:
ah ok, i get it: adding a sleep before the family join reproduces an error. thanks for explanation.